packages/block-library/src/terms-query/edit/inspector-controls/index.js

```diff
@@ -41,20 +45,57 @@ export default function TermsQueryInspectorControls( {
 		include,
 	} = termQuery;
 	const dropdownMenuProps = useToolsPanelDropdownMenuProps();
+	const [ hasAutoSetInherit, setHasAutoSetInherit ] = useState( false );
 
 	const taxonomies = usePublicTaxonomies();
 
+	const templateContext = useSelect(
+		( select ) => {
+			if ( ! templateSlug ) {
+				return null;
+			}
+			const { getTaxonomy } = select( coreStore );
+			return parseTemplateSlugWithValidation( templateSlug, getTaxonomy );
+		},
+		[ templateSlug ]
+	);
+
 	const isTaxonomyHierarchical = taxonomies.find(
 		( _taxonomy ) => _taxonomy.slug === taxonomy
 	)?.hierarchical;
 	const inheritQuery = !! inherit;
+
 	// Display the inherit control when we're in a taxonomy-related
 	// template (category, tag, or custom taxonomy).
 	const displayInheritControl =
+		templateContext?.taxonomy ||
 		[ 'taxonomy', 'category', 'tag', 'archive' ].includes( templateSlug ) ||
 		templateSlug?.startsWith( 'taxonomy-' ) ||
 		templateSlug?.startsWith( 'category-' ) ||
 		templateSlug?.startsWith( 'tag-' );
+
+	// Auto-set inherit mode and taxonomy when we have a valid template context taxonomy.
+	useEffect( () => {
```

Comment:
I believe this should be explored separately from this PR. Personally I don't think we should have this logic in this block. We can fall in `undo` traps and make it impossible for such a template to have two instances of this block, if they want an `inherit` one and a `custom` one too.

Similar thing had been explored for Query Loop in the past and dismissed for the above reasons.